### PR TITLE
Improve duplicate field error message

### PR DESCRIPTION
**Pull request by pmantica1**

### CODE REVIEW

File: graphql_compiler/compiler/compiler_frontend.py [hunk 1]
```diff
@@ -173,7 +173,9 @@ def _get_fields(ast):
             # If we ever allow repeated field names,
             # then we have to change the Location naming scheme to reflect the repetitions
             # and disambiguate between Recurse and Traverse visits to a Location.
-            raise GraphQLCompilationError(u'Encountered repeated field name: {}'.format(name))
+            raise GraphQLCompilationError(u'Encountered repeated field name: {}. To use '
+                                          u'multiple directives on a field, write the other '
+                                          u'directives directly below the first one.'.format(name))
```
Comment (obi1kenobi) [inline graphql_compiler/compiler/compiler_frontend.py]: I think this is solving a real problem that is worth solving. We just have to be mindful that our solution doesn't cause more damage than good.

There are other situations where this error can happen, it's not just multiple directives. In principle, the following will raise this error as well:
```
{
    Animal {
        name
        name
        alias @output(out_name: "alias")
    }
}
```

If I were to write the above, and then see this error message in response, that would be very confusing.

Also, "directly below" is misleading and unnecessarily strict -- not all editors have "directly below" as an option (e.g. consider the default Flask-JSONRPC frontend). How about "adjacent to" instead?

Comment (bojanserafimov) [inline graphql_compiler/compiler/compiler_frontend.py]: This advice applies only to duplicate property fields, and is very misleading for duplicate vertex fields. Can you distinguish which one it is? Also, say `after` the first one, not `below`.

File: graphql_compiler/compiler/compiler_frontend.py [hunk 2]
```diff
@@ -173,7 +173,15 @@ def _get_fields(ast):
...
-            raise GraphQLCompilationError(u'Encountered repeated field name: {}'.format(name))
+            raise GraphQLCompilationError(u'Encountered repeated field name: {}. If you are '
+                                          u'attempting to specify multiple directives on a single '
+                                          u'property field, one way to do so is to follow the '
+                                          u'format: propertyField directive1 directive2 ...'
+                                          u'However, note that all whitespace sequences, '
+                                          u'(e.g. spaces, tabs and newline characters),  are '
+                                          u'treated in the same way as delimiters in GraphQL, '
+                                          u'so feel free to space out directives in a different '
+                                          u'fashion.'.format(name))
```
Comment (bojanserafimov) [inline graphql_compiler/compiler/compiler_frontend.py]: This is way too technical, just to explain that if the line is too long they can break it up.

File: graphql_compiler/compiler/compiler_frontend.py (hunk 1 above)
Comment (bojanserafimov) [reply graphql_compiler/compiler/compiler_frontend.py]: Can you actually distinguish with an if statement and separate error message? I still think this message is adding confusion when there's duplicate vertex fields.

**Pull request merged**